ouroboros-network/src/Ouroboros/Network/DeltaQ.hs

```diff
@@ -0,0 +1,117 @@
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+
+module Ouroboros.Network.DeltaQ (
+    -- * GSV type
+    GSV(..),
+    Distribution,
+
+    -- * DeltaQ calculations
+    SizeInBytes,
+    DeltaQ,
+    gsvLeadingEdgeArrive,
+    gsvTrailingEdgeDepart,
+    gsvTrailingEdgeArrive,
+    deltaqQ99thPercentile,
+    deltaqProbabilityMassBeforeDeadline,
+
+    -- * Bi-directional GSV
+    PeerGSV(..),
+    gsvRequestResponseDuration,
+  ) where
+
+import           Data.Semigroup ((<>))
+import           Control.Monad.Class.MonadTimer (TimeMeasure, Duration)
+
+
+--
+-- Sketch of API for GSV and DeltaQ
+--
+
+data Distribution n = Distribution n
+-- This is a totally bogus representation. It's just a PoC.
+-- This says that there's a single value that it takes with probability 1.
+
+instance Num n => Semigroup (Distribution n) where
+  (<>) = convolveDistribution
+
+convolveDistribution :: Num n => Distribution n -> Distribution n -> Distribution n
+convolveDistribution (Distribution d) (Distribution d') = Distribution (d+d')
+-- Again, totally bogus.
+
+shiftDistribution :: Num n => n -> Distribution n -> Distribution n
+shiftDistribution n (Distribution d) = Distribution (n+d)
+-- Again, totally bogus.
+
+data GSV t = GSV (Duration t)                -- G as seconds
```

Comment:
You should describe what the G, S and V is. The unit gives a hint but that isn't enough.

Reply:
see appendix A of https://www.ofcom.org.uk/__data/assets/pdf_file/0024/71682/traffic-management-detection.pdf that is a network centric introduction

Reply:
Maybe it makes sense to put a link to it somewhere in a comment. For @MarcFontaine and future generations to study this code :)

Reply:
I think that the definitions of G, S and V should be extracted from that document and included in a comment.

Reply:
Yep, I should document this.

Reply:
Done.